src/fs.c

```diff
@@ -455,32 +464,16 @@ static int luv_fs_unlink(lua_State* L) {
 static int luv_fs_write(lua_State* L) {
   luv_ctx_t* ctx = luv_context(L);
   uv_file file = luaL_checkinteger(L, 1);
-  uv_buf_t buf;
-  int64_t offset;
-  int ref;
-  uv_fs_t* req;
-  size_t count;
-  uv_buf_t *bufs = NULL;
-
-  if (lua_istable(L, 2)) {
-    bufs = luv_prep_bufs(L, 2, &count);
-    buf.base = NULL;
-  }
-  else if (lua_isstring(L, 2)) {
-    luv_check_buf(L, 2, &buf);
-    count = 1;
-  }
-  else {
-    return luaL_argerror(L, 2, "data must be string or table of strings");
-  }
-
-  offset = luaL_checkinteger(L, 3);
-  ref = luv_check_continuation(L, 4);
-  req = (uv_fs_t*)lua_newuserdata(L, sizeof(*req));
+  int64_t offset = luaL_checkinteger(L, 3);
+  int ref = luv_check_continuation(L, 4);
+  uv_fs_t* req = (uv_fs_t*)lua_newuserdata(L, sizeof(*req));
   req->data = luv_setup_req(L, ctx, ref);
-  req->ptr = buf.base;
-  ((luv_req_t*)req->data)->data = bufs;
-  FS_CALL(write, req, file, bufs ? bufs : &buf, count, offset);
+  size_t count;
+  uv_buf_t* bufs = luv_check_bufs(L, 2, &count, (luv_req_t*)req->data);
```

Comment:
Can't do that, since `luv_check_bufs` uses `(luv_req_t*)req->data` which is created by `req->data = luv_setup_req(L, ctx, ref);` which depends on `int ref = luv_check_continuation(L, 4);`.

This is also true for all the other `luv_check_bufs` calls.